node/pkg/watchers/algorand/watcher_test.go

```diff
@@ -0,0 +1,83 @@
+package algorand
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/certusone/wormhole/node/pkg/common"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"go.uber.org/zap"
+)
+
+// TODO: grab these from const file somewhere?
+const (
+	IDX_URL   = "https://testnet-idx.algonode.cloud"
```

Comment:
Should a go test be accessing external endpoints? Would it be better to either hard code the data here are directly invoke gatherObservations, or move this test into a ci test?

Reply:
@barnjamin other tests use a hardcoded response as seen in this PR https://github.com/wormhole-foundation/wormhole/pull/3080. could you do something like that?

Reply:
added a flat file with a json payload containing the transaction with nested inner transactions.

had to do some gross stuff for the log message, detailed in the comments